server/tests/test_api_get_lecture.py

```diff
@@ -0,0 +1,89 @@
+import json
+
+import dateutil.parser
+
+from server.tests.base import BaseTestCase
+from server.models import db, Lecturer, Course, Lecture, Comment
+
+
+class GetLectureListApiTest(BaseTestCase):
+    def setUp(self):
+        super(GetLectureListApiTest, self).setUp()
+
+        self.simon = Lecturer('simon', '1234', 'Simon', 'McCallum')
+        db.session.add(self.simon)
+
+        self.imt3601 = Course('IMT3601 - Game Programming', self.simon)
+        db.session.add(self.imt3601)
+
+        self.imt3601_l1 = Lecture('Lecture 1', self.imt3601)
+        db.session.add(self.imt3601_l1)
+
+        self.frode = Lecturer('frode', '1234', 'Frode', 'Haug')
+        db.session.add(self.frode)
+
+        self.imt1031 = Course('IMT1031 - Grunnleggende Programmering', self.frode)
+        db.session.add(self.imt1031)
+
+        self.imt1031_l1 = Lecture('Lecture 1', self.imt1031)
+        db.session.add(self.imt1031_l1)
+        
+        self.imt1031_l2 = Lecture('Lecture 2', self.imt1031)
+        db.session.add(self.imt1031_l2)
+        
+        self.imt1031_l3 = Lecture('Lecture 3', self.imt1031)
+        db.session.add(self.imt1031_l3)
+
+        db.session.commit()
+
+    def test_success(self):
+        res = self.client.get('/api/0/lectures')
+        self.assert200(res)
+
+    def test_response_format(self):
+        res = self.client.get('/api/0/lectures')
+        self.assertEqual(res.headers['Content-Type'], 'application/json')
+
+    def test_response_is_list(self):
+        res = self.client.get('/api/0/lectures')
+        response = json.loads(res.data.decode('utf-8'))
+        self.assertIsInstance(response, list)
+
+    def test_content(self):
+        res = self.client.get('/api/0/lectures')
+        response = json.loads(res.data.decode('utf-8'))
+
+        self.assertEqual(len(response), 4)
+        self.assertEqual(response[0]['id'], self.imt3601_l1.id)
+        self.assertEqual(response[1]['id'], self.imt1031_l1.id)
+        self.assertEqual(response[2]['id'], self.imt1031_l2.id)
+        self.assertEqual(response[3]['id'], self.imt1031_l3.id)
+        self.assertEqual(response[0]['name'], self.imt3601_l1.name)
+        self.assertEqual(response[1]['name'], self.imt1031_l1.name) 
+        self.assertEqual(response[2]['name'], self.imt1031_l2.name)
+        self.assertEqual(response[3]['name'], self.imt1031_l3.name)
```

Comment:
We don't specify the order in which API should return those lectures, so you can just create set of all lecture names and ids and check that they are equal:

Something like this:
In __init__:
```
self.lecture_names = {imt3601_l1.name, imt1031_l1.name...}
self.lecture_ids = {imt3601_l1.id, imt1031_l1.id, ...}
```
And here:
```
lecture_names_res = {lecture['name'] for lecture in response}
lecture_ids_res = {lecture['id'] for lecture in response}

self.assertEqual(lecture_names_res, self.lecture_names
```